sklearn/decomposition/pca.py

```diff
@@ -396,14 +406,15 @@ def _fit(self, X):
                 self._fit_svd_solver = 'full'
             elif n_components >= 1 and n_components < .8 * min(X.shape):
                 self._fit_svd_solver = 'randomized'
+            # need to add 'lobpcg' here
```

Comment:
Do you have a good heuristic for deciding lobpcg is most appropriate?

Reply:
Not at this point, otherwise I would already put something specific here.

Reply:
https://github.com/scikit-learn/scikit-learn/pull/12319#discussion_r275685686 fixed in
https://github.com/scikit-learn/scikit-learn/pull/12319/commits/70cf7ca7ae38af32a175f37732e183265e38062a